src/libstd/fs.rs

```diff
@@ -844,6 +844,8 @@ pub fn symlink_metadata<P: AsRef<Path>>(path: P) -> io::Result<Metadata> {
 
 /// Rename a file or directory to a new name.
 ///
+/// This will not work if the new name is on a different filesystem.
+///
```

Comment:
Yeah, filesystem is rather ambiguous. Maybe @steveklabnik should simply say mount point?